data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/actions/CombineAggregateAction.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate.actions;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateAction;
+import com.amazon.dataprepper.plugins.processor.aggregate.AggregateActionResponse;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class CombineAggregateAction implements AggregateAction {
+    static final String COMBINED_EVENT_TYPE = "combined_event";
+
+    @Override
+    public AggregateActionResponse handleEvent(Event event, Map<Object, Object> groupState) {
+        if (groupState != null)
+            groupState.putAll(event.toMap());
```

Comment:
The results of this aggregation seems more like an overwrite than a combination. Order of events received will impact the output. Is this the expected behavior?

Reply:
I was a little confused on this too but @dlv discussed it. The idea is that there are the following logs which are grouped by the identification keys `sourceIp`, `destinationIp` and `port`. Sometimes different pieces of these logs will come in over time, so let's say you have this

` { sourceIp: 123, destinationIp: 456, port: 100, responseStatus: 200 } `

and then a little later you get 

` { sourceIp: 123, destinationIp: 456, port: 100, messageSize: 1,000 } `

then you want to combine these two together to have both messageSize and responseStatus

` { sourceIp: 123, destinationIp: 456, port: 100, messageSize: 1,000, responseStatus: 200 } `

I believe this scenario was framed to David by Cargill, so that is where the idea came from. Of course, we will get their feedback once we have some more implementation, and we can discuss with them if this is what they had in mind for their AggregateActions.